newsroom/search/service.py

```diff
@@ -550,14 +550,16 @@ def prefill_search_highlights(self, search, req):
             )
 
             for field in fields_to_highlight:
-                search.source["highlight"]["fields"][field] = {
+                highlight_search.source["highlight"]["fields"][field] = {
                     "number_of_fragments": 0,
-                    "highlight_query": {"bool": {"should": should_queries}},
+                    "highlight_query": {"bool": {"filter": highlight_search.query["bool"]["filter"]}},
                 }
 
-            search.source["highlight"]["pre_tags"] = ['<span class="es-highlight">']
-            search.source["highlight"]["post_tags"] = ["</span>"]
-            search.source["highlight"]["require_field_match"] = False
+            highlight_search.source["highlight"]["pre_tags"] = ['<span class="es-highlight">']
+            highlight_search.source["highlight"]["post_tags"] = ["</span>"]
+            highlight_search.source["highlight"]["require_field_match"] = False
```

Comment:
not sure if we need this, it should probably only highlight those requested fields

Reply:
yep